src/ProjectOrigin.WalletSystem.IntegrationTests/REST/TransfersControllerTests.cs

```diff
@@ -178,14 +181,82 @@ public async Task TransferCertificate_Unauthorized()
         // Act
         var result = await controller.TransferCertificate(
             provider.GetRequiredService<ITestHarness>().Bus,
+            _unitOfWork,
             _fixture.Create<TransferRequest>());
 
         // Assert
         result.Result.Should().BeOfType<UnauthorizedResult>();
     }
 
     [Fact]
-    public async void TransferCertificate_PublishesCommand()
+    public async Task GetTransferStatus_Unauthorized()
+    {
+        var controller = new TransfersController();
+
+        await using var provider = new ServiceCollection()
+            .AddMassTransitTestHarness(x =>
+            {
+            })
+            .BuildServiceProvider(true);
+
+        var result = await controller.GetTransferStatus(_unitOfWork, Guid.NewGuid());
+
+        result.Result.Should().BeOfType<UnauthorizedResult>();
+    }
+
+    [Fact]
+    public async Task GetTransferStatus_NotFound()
+    {
+        var subject = _fixture.Create<string>();
+        var controller = new TransfersController
+        {
+            ControllerContext = CreateContextWithUser(subject)
+        };
+
+        await using var provider = new ServiceCollection()
+            .AddMassTransitTestHarness(x =>
+            {
+            })
+            .BuildServiceProvider(true);
+
+        var result = await controller.GetTransferStatus(_unitOfWork, Guid.NewGuid());
+
+        result.Result.Should().BeOfType<NotFoundResult>();
+    }
+
+    [Fact]
+    public async Task GetTransferStatus()
+    {
+        var subject = _fixture.Create<string>();
+        var controller = new TransfersController
+        {
+            ControllerContext = CreateContextWithUser(subject)
+        };
+
+        await using var provider = new ServiceCollection()
+            .AddMassTransitTestHarness(x =>
+            {
+            })
+            .BuildServiceProvider(true);
+
+        var transferRequestId = Guid.NewGuid();
+        var status = new Server.Models.RequestStatus
```

Comment:
Why this namespace?

Reply:
Because the model and the dto is called the same